projects/storefrontlib/cms-components/navigation/scroll-to-top/scroll-to-top.component.ts

```diff
@@ -53,25 +57,65 @@ export class ScrollToTopComponent implements OnInit {
     this.setConfig();
   }
 
-  protected setConfig(): void {
-    this.componentData.data$.pipe(take(1)).subscribe((data) => {
-      this.scrollBehavior = data.scrollBehavior ?? this.scrollBehavior;
-      this.displayThreshold = data.displayThreshold ?? this.displayThreshold;
-    });
+  @HostListener('window:scroll', ['$event'])
+  onScroll(): void {
+    this.switchDisplay();
   }
 
   /**
    * Scroll back to the top of the page and set focus on top most focusable element.
    */
-  scrollToTop(): void {
-    // Focus first focusable element within the html body
-    this.selectFocusUtility
-      .findFirstFocusable(this.winRef.document.body, { autofocus: '' })
-      ?.focus();
-
+  scrollToTop(event: MouseEvent): void {
     this.window?.scrollTo({
       top: 0,
       behavior: this.scrollBehavior,
     });
+
+    this.clickEventSource = event.detail;
+  }
+
+  focusOut(): void {
+    if (this.display) {
+      this.switchDisplay();
+    }
+  }
+
+  @HostListener('document:keydown', ['$event'])
+  handleKeyboardEvent(event: KeyboardEvent): void {
+    if (
+      document.activeElement === this.button.nativeElement &&
+      this.clickEventSource !== undefined &&
+      event.key === 'Tab' &&
+      !event.shiftKey
+    ) {
+      event.preventDefault();
+
+      // Focus first focusable element within the html body
+      this.selectFocusUtility
+        .findFirstFocusable(this.winRef.document.body, { autofocus: '' })
+        ?.focus();
+    }
+  }
+
+  protected setConfig(): void {
+    this.componentData.data$.pipe(take(1)).subscribe((data) => {
+      this.scrollBehavior = data.scrollBehavior ?? this.scrollBehavior;
+      this.displayThreshold = data.displayThreshold ?? this.displayThreshold;
+    });
+  }
+
+  private switchDisplay(): void {
```

Comment:
Maybe we could comment what this function `switchDisplay()` is doing?

Reply:
I've added comments and refactored for readability.